### ISSUE TITLE
GetTwoFactorAuthenticationUserAsync();

### ISSUE THREAD

**Issue by nicowitteman** [damienbod/AspNetCoreIdentityFido2Mfa#9]
await _signInManager.GetTwoFactorAuthenticationUserAsync() always returns null in release version, while it does function in development mode. I do see a cookie named Identity.TwoFactorUserId being set after the first step of logging in.

My environment is .NET 5.0

**Comment by damienbod** [comment]
Hi @nicowitteman 

This should work, I'll have a look

Greetings Damien

**Comment by nicowitteman** [comment]
Hello Damien, I forgot to write that I made some changes in the code make the solution work in a subfolder (which was quite a trial and error effort). Could there be a relation? Nico

**Comment by nicowitteman** [comment]
I'm afraid my diagnosis was incorrect: it does work on localhopst in both dev and release modes, but it fails when published. I publish in self-contained win-x64, running out-of-process. Any clue? Nico

**Comment by damienbod** [comment]
This should work as well, the app.setting for the domain needs to mathc with your deployment

Greetings Damien

**Comment by nicowitteman** [comment]
It does match, Damien, or it would not work at all. I have serverDomain and origin matching the deployment. That is to say, both without the folder name I deployed into, which is the same as I do for localhost. I believe serverName can be anything, and I did not touch timestampDriftTolerance (the server time is correct) or MDSAccessKey.

**Comment by nicowitteman** [comment]
It appears I've been looking in the wrong direction, Damien. Everything works now, until I update the app. Is it logical that all FIDO stored credentials are invalidated by an update of the binaries of my application, even if appsettings are unchanged? How should we handle this as developers?

**Comment by nicowitteman** [comment]
It looks like something is going wrong in the AuthenticationHandler. Can't figure it out yet. It is just not stable - it works at times, and then suddenly it doesn't.

**Comment by nicowitteman** [comment]
Hi Damien, let's close this issue - it is not in your code. At times I get an error from navigator.credentials.get, about the operation not being allowed. In Edge, a reference is shown to https://www.w3.org/TR/webauthn-2/#sctn-privacy-considerations-client.

**Comment by damienbod** [comment]
@nicowitteman Couls you post a comment when you find a solution? When be interesting

Greetings Damien

**Comment by nicowitteman** [comment]
Hi Damien, I have new insights: in the cases SignInManager returns null and therefore the login fails, the preceding /mfassertionOptions POST had not included the Identity.TwoFactorUserId cookie, even though the browser shows it as being present. This happens mostly using the Firefox browser, but also Chrome and Edge fail occasionally. Safari on my iPhone on the other hand always works as expected. Does this behaviour ring a bell?

**Comment by damienbod** [comment]
@nicowitteman Have not experienced this, thanks for the feedback.

**Comment by nicowitteman** [comment]
I think I got it, finally: since I deployed my solution in a folder, and the Identity.TwoFactorUserId-cookie is stored with the path=folder, no cookie was found when the url wasn't all lowercase. This, and case-sensitivity of cookies! So it worked when I went to /fido2mfa/login, but it didn't when I went to /Fido2MFA/login. Now I force all url's to lowercase in web.config, the cookie is created with the lowercase pathname and no cookie confusion is possbile anymore.
